src/borg/remote.py

```diff
@@ -81,8 +83,19 @@ def get_limited_unpacker(kind):
         args.update(dict(max_array_len=LIST_SCAN_LIMIT,  # result list from repo.list() / .scan()
                          max_map_len=100,  # misc. result dicts
                          ))
+    elif kind == 'manifest':
+        args.update(dict(use_list=True,  # default value
+                         max_array_len=100,  # ITEM_KEYS ~= 22
+                         max_map_len=MAX_ARCHIVES,  # list of archives
+                         max_str_len=1000,  # archive name
```

Comment:
well, the intention of this limitation is to avoid attacks, not to check length for other parts of the code. i'll add some more asserts to write().